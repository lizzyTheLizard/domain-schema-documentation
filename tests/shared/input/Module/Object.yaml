$id: /Module/Object.yaml
title: Object
description: This is an integration test for the application
type: object
x-schema-type: Aggregate
examples:
  - id: "1"
    name: "Test"
properties:
  id:
    type: string
  name:
    type: string
