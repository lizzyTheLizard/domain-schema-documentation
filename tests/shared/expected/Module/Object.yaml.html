<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.84.0">
    <title>Test Application</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/navbar-static/">
    <link rel=stylesheet href=https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/css/pretty-print-json.css>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="theme-color" content="#7952b3">
    <style>
    .alert ul {
      margin-bottom: 0;
    }
    .mermaid{
      height: 300px;
      border: 1px solid #000;
    }
    .mermaid pre {
      display: none;
    }
    section {
      margin-bottom: 50px;
    }
    .badges {
      margin-bottom: 15px;
    }
    .json {
      border: 1px solid black;
      background-color: lightgray;
      padding: 10px;
     }
    </style>
  </head>
  <body>
    <main class="container">
      <nav aria-label="breadcrumb">
        <ol class="breadcrumb">
          <li class="breadcrumb-item"><a href="../index.html">Application</a></li>
          <li class="breadcrumb-item"><a href="./index.html">Module</a></li>
          <li class="breadcrumb-item active" aria-current="page">Object</li>
        </ol>
      </nav>
      
      <h1>Object</h1>
      
      
      <section>
        <p>A simple object</p>
        <div class="mermaid">
          <pre class="mermaidDefinition">
            classDiagram
      namespace Module {
        class _Module_Object_yaml["Object"]
        class SubInterface["SubInterface"]
        class SubInterfaceOneOf3["SubInterfaceOneOf3"]
        class SubInterfaceOneOf2["SubInterfaceOneOf2"]
        class SubEnum["SubEnum"]
        class SubObject["SubObject"]
        class _Module_Interface_yml["Interface"]
      }
      namespace Module 2 {
        class _Module2_Object2_yaml["Object 2"]
      }
      _Module2_Object2_yaml o-- _Module_Object_yaml :reference1
      _Module2_Object2_yaml ..> _Module_Object_yaml :reference2
      _Module_Object_yaml o-- SubObject :subObject
      _Module_Object_yaml ..> SubEnum :subEnum
      _Module_Object_yaml o-- SubInterface :subInterface
      SubInterface <|-- _Module_Object_yaml 
      SubInterface <|-- SubInterfaceOneOf2 
      SubInterface <|-- SubInterfaceOneOf3 
      SubInterfaceOneOf3 <|-- _Module_Object_yaml 
      SubInterfaceOneOf3 <|-- _Module_Interface_yml 
      click _Module_Object_yaml href "./Object.yaml.html" "Object"
      click SubInterface href "./Object.yaml.html" "Object"
      click SubInterfaceOneOf3 href "./Object.yaml.html" "Object"
      click SubInterfaceOneOf2 href "./Object.yaml.html" "Object"
      click SubEnum href "./Object.yaml.html" "Object"
      click SubObject href "./Object.yaml.html" "Object"
      click _Module2_Object2_yaml href "../Module2/Object2.yaml.html" "Object 2"
      click _Module_Interface_yml href "./Interface.yml.html" "Interface"
          </pre>
        </div>
      </section>
      
      
      
      <section>
        <h2>Properties</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>id*</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>name*</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>subObject</td>
              <td><a href="#SubObject">SubObject</a></td>
              <td></td>
            </tr>
            <tr>
              <td>subEnum</td>
              <td><a href="#SubEnum">SubEnum</a></td>
              <td></td>
            </tr>
            <tr>
              <td>subInterface</td>
              <td><a href="#SubInterface">SubInterface</a></td>
              <td></td>
            </tr>
            <tr>
              <td>(other)</td>
              <td>Integer</td>
              <td>Additional Properties</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
        <h2>Examples</h2>
        <pre class="json">
        {"id":"1","name":"Test"}
        </pre>
      </section>
      
      <h2>Subschemas</h2>
      <a id="SubInterface"></a><h3>Interface SubInterface</h3>
      
      
      <section>
        <table class="table">
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>key</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>value</td>
              <td>String</td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
      Implemented by
        <ul>
          <li><a href="">Object</a></li>
          <li><a href="#SubInterfaceOneOf2">SubInterfaceOneOf2</a></li>
          <li><a href="#SubInterfaceOneOf3">SubInterfaceOneOf3</a></li>
        </ul>
      </section>
      <a id="SubInterfaceOneOf3"></a><h3>Interface SubInterfaceOneOf3</h3>
      
      
      
      <section>
      Implemented by
        <ul>
          <li><a href="">Object</a></li>
          <li><a href="./Interface.yml.html">Interface</a></li>
        </ul>
      </section>
      <a id="SubInterfaceOneOf2"></a><h3>Class SubInterfaceOneOf2</h3>
      
      
      <section>
        <table class="table">
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>key</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>value</td>
              <td>String</td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <a id="SubEnum"></a><h3>Enum SubEnum</h3>
      
      <section>
        <table class="table">
          <thead>
            <tr>
              <th>Enum</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>A</td>
              <td>Value A</td>
            </tr>
            <tr>
              <td>B</td>
              <td>Value B</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      
      <a id="SubObject"></a><h3>Class SubObject</h3>
      
      
      <section>
        <table class="table">
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>key</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>value</td>
              <td>String</td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
      
      
      
    </main>
    <script src="https://cdn.jsdelivr.net/npm/svg-pan-zoom@3.5.0/dist/svg-pan-zoom.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/pretty-print-json.min.js"></script>
    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';

      function prettifyJson(){
        const elements = document.querySelectorAll('.json');
        for(const element of elements) {
          element.innerHTML = prettyPrintJson.toHtml(JSON.parse(element.innerText));
        }
      }

      function drawDiagrams() {
        const elements = document.querySelectorAll('.mermaid');
        elements.forEach((element, i) =>drawDiagram(element, i));
      }

      async function drawDiagram(element, i) {
        const graphDefinition = element.querySelector(".mermaidDefinition")
        const {svg} = await mermaid.render('mySvg_' + i, graphDefinition.innerText);
        const preview = createMermaidPreview(element, svg);
        const modal = createMermaidModal();
        //Do this with a slight delay to take link in mermaid diagram take precedence
        element.addEventListener('click', () => setTimeout(() => showMermaidModal(preview,modal), 30));
      }

      function createMermaidPreview(element, svg) {
        const preview = document.createElement('div');
        preview.style.height = '100%';
        preview.style.width = '100%';
        preview.innerHTML = svg.replace(/[ ]*max-width:[ 0-9\.]*px;/i , '');
        preview.querySelector("svg").style.height = '100%';
        preview.querySelector("svg").style.width = '100%';
        element.appendChild(preview);
        return preview;
      }

      function createMermaidModal() {
        const modal = document.createElement('div');
        modal.innerHTML =
            '<div class="modal" tabindex="-1">\n' +
            '  <div class="modal-dialog modal-fullscreen">\n' +
            '    <div class="modal-content">\n' +
            '      <div class="modal-header">\n' +
            '        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>\n' +
            '      </div>\n' +
            '      <div class="modal-body"></div>\n' +
            '    </div>\n' +
            '  </div>\n' +
            '</div>';
        document.body.appendChild(modal);
        return modal;
      }

      function showMermaidModal(preview, modal) {
        const modalBody = modal.querySelector('.modal-body');
        modalBody.innerHTML = preview.innerHTML
        const modalDiv = modal.querySelector('.modal');
        const svg = modalBody.querySelector('svg');
        new bootstrap.Modal(modalDiv).show();
        svgPanZoom(svg, {
          zoomEnabled: true,
          controlIconsEnabled: true,
          fit: true,
          center: true
        })
      }

      mermaid.initialize({ startOnLoad: false,securityLevel: 'loose'});
      drawDiagrams();
      prettifyJson();
    </script>
  </body>
</html>
